### PR TITLE
Mnt drop py36

### LINKED ISSUE #17652: NEP 29 : Stop support fro Python 3.6 soon ?
[NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html) numpy/numpy#14086 suggest dropping Python 3.6 support soon.  Matplotlib is a co-signer.

Dask is [considering it](https://github.com/dask/community/issues/66), will matplotlib consider following nep 29 and remove Python 3.6 from official support?

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Drop support for py36

closes #17652

## PR Checklist

This should not go in until after we branch v3.3.x off of master and still needs an API change note added (but I want to let the dust settle on the renaming / moving associated with 3.3 before doing that.

### CODE REVIEW

**Comment by tacaswell** [comment]
I can squash this to one commit if we want.

**Comment by anntzer** [comment]
(prefer this being squashed)

**Comment by tacaswell** [comment]
I left each minimum version change as its own commit but squashed the CI thrashing.

File: .appveyor.yml [hunk 1]
```diff
@@ -23,13 +23,14 @@ environment:
...
     # making things faster due to package caching.
-    - PYTHON_VERSION: "3.6"
-      CONDA_INSTALL_LOCN: "C:\\Miniconda36-x64"
+    - PYTHON_VERSION: "3.7"
+      CONDA_INSTALL_LOCN: "C:\\Miniconda37-x64"
       TEST_ALL: "no"
       EXTRAREQS: "-r requirements/testing/travis_extra.txt"
-    - PYTHON_VERSION: "3.7"
+    - PYTHON_VERSION: "3.8"
       CONDA_INSTALL_LOCN: "C:\\Miniconda37-x64"
```
Comment (QuLogic) [inline .appveyor.yml:31]:
```suggestion
      CONDA_INSTALL_LOCN: "C:\\Miniconda38-x64"
```

File: .appveyor.yml [hunk 2]
```diff
@@ -55,19 +56,15 @@ install:
   # For building, use a new environment
   - conda create -q -n test-environment python=%PYTHON_VERSION% tk
   - activate test-environment
+  # pull pywin32 from conda because on py38 there is something wrong with finding
```
Comment (QuLogic) [inline .appveyor.yml:59]: Is that because of wrong conda above?

File: .circleci/config.yml [hunk 3]
```diff
@@ -110,26 +110,6 @@ commands:
...
-    docker:
-      - image: circleci/python:3.6
-    steps:
-      - checkout
-
-      - apt-install
-      - fonts-install
-      - pip-install
-      - deps-install:
-          numpy_version: "==1.13.0"
```
Comment (QuLogic) [inline .circleci/config.yml:123]: Should leave a minimum NumPy in 3.7 build?

File: .appveyor.yml (hunk 1 above)
Comment (tacaswell) [reply .appveyor.yml:31]: This doesn't work unless we update the windows image we are using.

File: .appveyor.yml (hunk 2 above)
Comment (tacaswell) [reply .appveyor.yml:59]: I don't think so, we are still ending up with a py38 env and it should be de-coupled from Python version of the "base" conda enivorment.  I think this has to do with Python tightening up its rules about how dlls are discovered.  There are a bunch of related issues on the pywin32 issues tracker.

File: .circleci/config.yml (hunk 3 above)
Comment (tacaswell) [reply .circleci/config.yml:123]: We have a test with the minimum numpy and friends on travis.  Are we worried about the docs having issues when the tests pass on a given numpy version?

Comment (QuLogic) [reply .circleci/config.yml:123]: Sometimes examples do weird things, so not sure. We could just say docs must be built on newer versions?

Comment (tacaswell) [reply .circleci/config.yml:123]: well, if the docs _do_ find weird behavior with old versions of numpy that is identifying a gap in our testing and is something we should fix 2x...

Comment (tacaswell) [reply .circleci/config.yml:123]: I put back a py38 + numpy1.16 build.

Comment (QuLogic) [reply .circleci/config.yml:123]: Circle maybe doesn't like that.

**Comment by QuLogic** [comment]
LGTM, but waiting on the API change note.

**Pull request merged**